packages/core/test/financial-templates/common/financial-product-libraries/long-short-pair-libraries/SimpleSuccessTokenLongShortPairFinancialProductLibrary.js

```diff
@@ -0,0 +1,99 @@
+
+// Tested Contract
+const SimpleSuccessTokenLongShortPairFinancialProductLibrary = getContract(
+  "SimpleSuccessTokenLongShortPairFinancialProductLibrary"
+);
+
+const LongShortPairMock = getContract("LongShortPairMock");
+
+const { toWei, toBN } = web3.utils;
+const strikePrice = toWei("400");
+const collateralPerPair = toBN(toWei("1"));
+
+describe("SimpleSuccessTokenLongShortPairFinancialProductLibrary", function () {
+  let simpleSuccessTokenLSPFPL;
+  let lspMock;
+  let accounts;
+
+  before(async () => {
+    await runDefaultFixture(hre);
+    accounts = await hre.web3.eth.getAccounts();
+  });
+
+  beforeEach(async () => {
+    simpleSuccessTokenLSPFPL = await SimpleSuccessTokenLongShortPairFinancialProductLibrary.new().send({
+      from: accounts[0],
+    });
+    lspMock = await LongShortPairMock.new(
+      "1000000", // _expirationTimestamp
+      collateralPerPair // _collateralPerPair
+    ).send({ from: accounts[0] });
+  });
+  describe("Long Short Pair Parameterization", () => {
+    it("Can set and fetch valid strikes", async () => {
+      await simpleSuccessTokenLSPFPL.methods
+        .setLongShortPairParameters(lspMock.options.address, strikePrice)
+        .send({ from: accounts[0] });
+
+      const setStrike = await simpleSuccessTokenLSPFPL.methods
+        .longShortPairStrikePrices(lspMock.options.address)
+        .call();
+      assert.equal(setStrike.toString(), strikePrice);
+    });
+    it("Can not re-use existing LSP contract address", async () => {
+      await simpleSuccessTokenLSPFPL.methods
+        .setLongShortPairParameters(lspMock.options.address, strikePrice)
+        .send({ from: accounts[0] });
+
+      // Second attempt should revert.
+      assert(
+        await didContractThrow(
+          simpleSuccessTokenLSPFPL.methods
+            .setLongShortPairParameters(lspMock.options.address, strikePrice)
+            .send({ from: accounts[0] })
+        )
+      );
+    });
+    it("Can not set invalid LSP contract address", async () => {
+      // LSP Address must implement the `expirationTimestamp method.
+      assert(
+        await didContractThrow(
+          simpleSuccessTokenLSPFPL.methods
+            .setLongShortPairParameters(ZERO_ADDRESS, strikePrice)
+            .send({ from: accounts[0] })
+        )
+      );
+    });
+  });
+  describe("Compute expiry tokens for collateral", () => {
+    beforeEach(async () => {
+      await simpleSuccessTokenLSPFPL.methods
+        .setLongShortPairParameters(lspMock.options.address, strikePrice)
+        .send({ from: accounts[0] });
+    });
+    it("Lower than strike should return 0.5", async () => {
+      const expiryTokensForCollateral = await simpleSuccessTokenLSPFPL.methods
+        .percentageLongCollateralAtExpiry(toWei("300"))
+        .call({ from: lspMock.options.address });
+      assert.equal(expiryTokensForCollateral.toString(), toWei("0.5"));
+    });
+    it("Higher than strike correct value", async () => {
```

Comment:
a few additional tests and comments might be useful here. I know this is really simple but just to be consistent with the other libraries.

Reply:
I added some more comments to explain what's going on in here. If you think additional tests are needed, let me know.